vm/interpreter/src/default_runtime.rs

```diff
@@ -733,7 +737,23 @@ where
         x if x == *MULTISIG_ACTOR_CODE_ID => multisig::Actor.invoke_method(rt, method_num, params),
         x if x == *REWARD_ACTOR_CODE_ID => reward::Actor.invoke_method(rt, method_num, params),
         x if x == *VERIFREG_ACTOR_CODE_ID => verifreg::Actor.invoke_method(rt, method_num, params),
-        _ => Err(actor_error!(SysErrorIllegalActor; "no code for actor at address {}", to)),
+        x => {
+            if rt.registered_actors.contains(&x) {
+                match x {
+                    x if x == *PUPPET_ACTOR_CODE_ID => {
+                        puppet::Actor.invoke_method(rt, method_num, params)
+                    }
+                    x if x == *CHAOS_ACTOR_CODE_ID => {
+                        chaos::Actor.invoke_method(rt, method_num, params)
+                    }
+                    _ => Err(
+                        actor_error!(SysErrorIllegalActor; "no code for actor at address {}", to),
+                    ),
+                }
+            } else {
+                Err(actor_error!(SysErrorIllegalActor; "no code for actor at address {}", to))
```

Comment:
I would change the error message for this part. Maybe say that the actor isn't registered?

Reply:
technically, should be something to say that the registered actor cid doesn't exist (other way around)

Reply:
I agree with austin. i will push that change

Reply:
> technically, should be something to say that the registered actor cid doesn't exist (other way around)

True just wanted to distinguish the error messages on 754 and 750. Cuz its possible the chaos or puppet actor werent registered

Reply:
makes sense. change pushed